data/model/repo_mirror.py

```diff
@@ -527,6 +527,23 @@ def validate_rule(rule_type, rule_value):
         )
 
 
+def clean_rule(rule_type, rule_value):
```

Comment:
I'm not opposed to doing this but we should also do validation and input sanitation on the frontend (the endpoint). It would be nice to enforce the input we receive from API Requests and give the appropriate response instead of silently mutating their data.

Reply:
Agreed, we should always avoid "magical" behavior whenever possible. Moving validation to the API layer and adding helpful tips in the frontend keeps business logic out of our data layer.